src/xcode/ENA/ENA/Source/HTTPClient/Service/Models/Receive/ModelWithCache.swift

```diff
@@ -0,0 +1,50 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+protocol ModelWithCaching {
+	var isCached: Bool { get set }
+}
+
+/**
+ Generel model to encapsulate the real model, which has to fullfill CBORDecoding. The main purpose of this model is to store a property if the real model returned from the service is from the cache or freshly fetched. Normally you cannot differentiate here.
+ */
+struct ModelWithCache<WrappedModel>: CBORDecoding, ModelWithCaching where WrappedModel: CBORDecoding {
```

Comment:
Why are there different protocols 'CBORDecoding' and 'ReceiveResource' ? They look quite the same. Could this be refactored to be simpler? Why isn't the 'ReceiveResource' also returning 'ModelDecodingError'? Would be more precise IMHO.

Reply:
I renamed the `CBORDecoding` to `CBORDecodable` and it's func to `make(with data: Data)` to clarify that `CBORDecodable` is for models and `ReceiveResource` for resources. This is the generic approach to map the generic resource onto the specific one (JSON, ProtoBuf, CBOR etc.)
`CBORDecodable` does not only decode the CBOR but also creates the model based on the CBOR.

At the end the protocols seems very similar but are for different things.